src/HttpClientFactory/Http/src/DependencyInjection/HttpClientBuilderExtensions.cs

```diff
@@ -485,6 +486,60 @@ public static IHttpClientBuilder AddTypedClient<TClient>(this IHttpClientBuilder
             return builder;
         }
 
+        /// <summary>
+        /// Sets the <see cref="Predicate{T}"/> which determines whether to log HTTP header value or redact before logging
+        /// </summary>
+        /// <param name="builder">The <see cref="IHttpClientBuilder"/>.</param>
+        /// <param name="isSensitiveHeader">The <see cref="Predicate{T}"/> which determines whether to log HTTP header value or redact before logging.</param>
+        /// <returns>The <see cref="IHttpClientBuilder"/>.</returns>
+        public static IHttpClientBuilder RedactLoggedHeaders(this IHttpClientBuilder builder, Predicate<string> isSensitiveHeader)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (isSensitiveHeader == null)
+            {
+                throw new ArgumentNullException(nameof(isSensitiveHeader));
+            }
+
+            builder.Services.Configure<HttpClientFactoryOptions>(builder.Name, options =>
+            {
+                options.IsSensitiveHeader = isSensitiveHeader;
+            });
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets the HTTP headers collection which values should be redacted before logging.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHttpClientBuilder"/>.</param>
+        /// <param name="redactedLoggedHeaderNames">The HTTP headers collection which values should not be logged.</param>
+        /// <returns>The <see cref="IHttpClientBuilder"/>.</returns>
+        public static IHttpClientBuilder RedactLoggedHeaders(this IHttpClientBuilder builder, ICollection<string> redactedLoggedHeaderNames)
```

Comment:
Suggestion: make this `IEnumerable<string>`. There's no real advantage of `ICollection<>` for this use case